python/lib/client/dmod/client/request_clients.py

```diff
@@ -175,6 +202,50 @@ async def delete_dataset(self, name: str, **kwargs) -> bool:
         self.last_response = await self.async_make_request(request)
         return self.last_response is not None and self.last_response.success
 
+    async def get_dataset_content_details(self, name: str, **kwargs) -> dict:
+        # TODO: later add things like created and last updated perhaps
+        query = DatasetQuery(query_type=QueryType.GET_DATASET_ITEMS)
+        request = DatasetManagementMessage(action=ManagementAction.QUERY, query=query, dataset_name=name)
+        self.last_response: DatasetManagementResponse = await self.async_make_request(request)
+        if self.last_response.success:
+            return self.last_response.data
+        else:
+            return {}
+
+    async def get_item_size(self, dataset_name: str, item_name: str) -> int:
+        query = DatasetQuery(query_type=QueryType.GET_ITEM_SIZE, item_name=item_name)
+        request = DatasetManagementMessage(action=ManagementAction.QUERY, query=query, dataset_name=dataset_name,
+                                           data_location=item_name)
+        self.last_response: DatasetManagementResponse = await self.async_make_request(request)
+        if self.last_response.success:
+            return self.last_response.data
+        else:
+            return -1
+
+    async def download_item_block(self, dataset_name: str, item_name: str, blk_start: int, blk_size: int) -> AnyStr:
+        """
+        Download a block/chunk of a given size and start point from a specified dataset file.
+
+        Parameters
+        ----------
+        dataset_name
+        item_name
+        blk_start
+        blk_size
+
+        Returns
+        -------
+        AnyStr
+            The downloaded block/chunk.
+        """
+        request = DatasetManagementMessage(action=ManagementAction.REQUEST_DATA, dataset_name=dataset_name,
+                                           data_location=item_name, blk_start=blk_start, blk_size=blk_size)
+        self.last_response: DatasetManagementResponse = await self.async_make_request(request)
+        if self.last_response.success:
+            return self.last_response.data
+        else:
+            return ''
```

Comment:
^^ ditto. I am thinking for non-dict return types, does it make sense to raise an exception here?